auth/server/base/src/main/java/org/wildfly/security/auth/callback/AuthenticationMechanismPropertiesCallback.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2023 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * A {@link javax.security.auth.callback.Callback} to inform a server authentication context of configured mechanism properties.
+ *
+ * As an informational {@code Callback} it is optional for the {@code CallbackHandler} to handle this.
+ *
+ */
+package org.wildfly.security.auth.callback;
+
+public class AuthenticationMechanismPropertiesCallback implements ExtendedCallback{
+
+    /**
+     * Property of the SASL EXTERNAL mechanism that indicates whether a certificate should be verified against the security domain.
+     */
+    private boolean saslSkipCertificateVerification;
+
+    public boolean getSaslSkipCertificateVerification() {
+        return this.saslSkipCertificateVerification;
+    }
+
+    public void setSaslSkipCertificateVerification(boolean skipCertificateVerification) {
```

Comment:
Just wondering if this setter is needed. I think our other callbacks that pass information to the SAC just set the required info in the constructor and then have a getter to retrieve the info.

Reply:
@fjuma I wanted to make this callback more generic, so in the future it could be used for different properties, so I did not want to force this particular property to the constructor. Should I change it?

Reply:
Maybe it could be a generic key / value properties that is exchanged?

Reply:
@darranl To use a "key/value" with this callback, I would need to import the _SASL_SKIP_CERTIFICATE_CERTIFICATION_  property in the ServerAuthenticationContext to check for its value. For that  I would have to add add a wildfly-elytron-sasl dependency to wildfly-elytron-auth-server

Reply:
Thanks @Skyllarr, making this callback generic so that it can be used in the future for additional things make sense. This looks good to me.